.github/workflows/build_commits.yaml

```diff
@@ -0,0 +1,29 @@
+name: build_commits
+
+
+on: [pull_request]
+
+jobs:
+  build_commits:
+    name: build commit code
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '^1.16'
+      - name: install llvm and libelf
+        run: |
+          sudo apt-get install llvm
+          sudo apt-get install libelf-dev
+      - name: git fetch commit code
+        run: |
+          git fetch origin pull/$PR/head:rebase-head
+          git checkout rebase-head
+        env:
+          PR: ${{ github.event.pull_request.number }}  
+      - name: build commits
+        run: git rebase --exec "make clean && make -j $(nproc)" rebase-head
```

Comment:
Unless `rebase-head` has a special meaning I'm not aware of, I don't think this is correct. We want to perform the rebase against the `master` commit on which the pull request is based. We can't just rebase against `master` because it may have been updated in the meantime and such a rebase could lead to conflicts.

Reply:
We first fetched the PR (git fetch origin pull/$PR/head:rebase-head) and then checked out this branch (git checkout rebase-head). This way we are not effecting the master branch. Does this makes sense?

Reply:
Ah, of course :facepalm: Thanks for the explanation @Ankurk99! :pray: